test/y2partitioner/device_description_test.rb

```diff
@@ -0,0 +1,210 @@
+        let(:scenario) { "bug_1145841.xml" }
+        let(:device_name) { "/dev/sdd1" }
+
+        it "includes the human readable filesystem type" do
+          expect(description).to include("Ext4")
+        end
+
+        it "includes 'Journal'" do
+          expect(description).to include("Journal")
+        end
+
+        it "includes the data device base name" do
+          expect(description).to include("BACKUP_R6")
+        end
+      end
+
+      context "but it is part of a multi-device filesystem" do
+        let(:scenario) { "btrfs2-devicegraph.xml" }
+        let(:device_name) { "/dev/sdb1" }
+
+        it "includes 'Part of'" do
+          expect(description).to include("Part of")
+        end
+
+        it "includes the filesystem name" do
+          expect(description).to include("btrfs")
+        end
+
+        it "includes the block device basename" do
+          expect(description).to include("sdb1")
+        end
+      end
+    end
+
+    context "when the device is not formatted" do
+      context "and it is an used LVM physical volume" do
+        let(:device_name) { "/dev/sdb1" }
+        let(:vg) { device.lvm_pv.lvm_vg }
+
+        context "in a volume group with name" do
+          it "includes 'PV of'" do
+            expect(description).to include("PV of")
+          end
+
+          it "includes the volume group name" do
+            expect(description).to include("vg0")
+          end
+        end
+
+        context "in a volume group with an empty name" do
+          before do
+            vg.vg_name = ""
+          end
+
+          it "returns 'PV of LVM'" do
+            expect(description).to eq("PV of LVM")
+          end
+        end
+      end
+
+      context "and it is an unused LVM physical volume" do
+        let(:scenario) { "unused_lvm_pvs.xml" }
+        let(:device_name) { "/dev/sda2" }
+
+        it "returns 'Unused LVM PV'" do
+          expect(description).to eq("Unused LVM PV")
+        end
+      end
+
+      context "and it is part of an MD RAID" do
+        let(:scenario) { "md_raid.yml" }
+        let(:device_name) { "/dev/sda2" }
+
+        it "includes 'Part of'" do
+          expect(description).to include("Part of")
+        end
+
+        it "includes the MD RAID name" do
+          expect(description).to include("md0")
+        end
+      end
+
+      context "and it is part of a bcache" do
+        let(:scenario) { "bcache1.xml" }
+        let(:device_name) { "/dev/vdc" }
+
+        it "includes 'Backing of'" do
+          expect(description).to include("Backing of")
+        end
+
+        it "includes the bcache name" do
+          expect(description).to include("bcache0")
+        end
+      end
+
+      context "and it is used as caching device in a bcache" do
+        let(:scenario) { "bcache1.xml" }
+        let(:device_name) { "/dev/vdb" }
+
+        it "returns 'Bcache cache'" do
+          expect(description).to include("Bcache caching")
```

Comment:
NP: I know you just moved the code, but the line above said "returns 'Bcache cache'"